service/matching/taskListManager.go

```diff
@@ -293,6 +303,25 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 	scope := metrics.MatchingTaskListMgrScope
 	timer := time.NewTimer(c.config.LongPollExpirationInterval)
 	defer timer.Stop()
+
+	pollerID, ok := ctx.Value(pollerIDKey).(string)
+	childCtx, cancel := context.WithCancel(ctx)
+	if ok {
```

Comment:
since the else{} part is only for backward compatibility, this code can be refactored to remove the else entirely i.e. move the `context.WithCancel(ctx)` within the `if ok` branch and initialize `childCtx = ctx` outside the if

Reply:
Done.